lua/telescope/actions/init.lua

```diff
@@ -244,8 +244,14 @@ end
 actions.git_checkout = function(prompt_bufnr)
   local selection = actions.get_selected_entry(prompt_bufnr)
   actions.close(prompt_bufnr)
-  local val = selection.value
-  os.execute('git checkout ' .. val)
+  os.execute('git checkout ' .. selection.value)
+end
+
+actions.git_checkout_current_buffer = function(prompt_bufnr)
+  local selection = actions.get_selected_entry(prompt_bufnr)
+  actions.close(prompt_bufnr)
+  local file_of_current_buffer = vim.fn.expand('%')
```

Comment:
We need to get the file here over opts. That is currently not possible but you can just add that to `lua/telescope/pickers.lua` where the metatable is definied. I kinda wanted to be able to access opts in actions as well. I am sure you could figure it out :) if not just ask